arrows/vtk/applets/fuse_depth.cxx

```diff
@@ -366,10 +382,29 @@ class fuse_depth::priv
         mesh_writer->AddInputDataObject(isosurface_mesh);
         mesh_writer->Write();
       }
+      else if ( extension == ".las" )
+      {
+        auto lgcs = vital::local_geo_cs();
+        read_local_geo_cs_from_file(lgcs, input_geo_origin_file);
```

Comment:
We probably want some sort of check here that the `lgcs` is valid.  I don't remember if `read_local_geo_cs_from_file` throws an exception or returns `false` on failure, but it would be good to catch failures somehow so we don't have a repeat of the odd behavior you observed previously with the empty geo origin file.

Reply:
The `read_local_geo_cs_from_file` function doesn't seem to respond at all on failure. The garbage data seemingly comes from [here](https://github.com/Kitware/kwiver/blob/master/vital/types/local_geo_cs.cxx#L46):
```c++
  {
    std::ifstream ifs(file_path);
    double lat, lon, alt;
->  ifs >> lat >> lon >> alt;
    lgcs.set_origin( geo_point( vector_3d(lon, lat, alt), SRID::lat_lon_WGS84) );
  }
```
I could add some semantic in the scope of `fuse_depth.cxx` checks to make sure that the data is within certain bounds 
```c++
  read_local_geo_cs_from_file(lgcs, input_geo_origin_file);
  auto coords = lcgs.origin().location();
  if (coords[0] < ... || coords[1] > ... || ...)
    ...
```
However, shouldn't this be solved in a modification of `read_local_geo_cs_from_file` in tandem with checks in  `write_local_geo_cs_to_file` to only create a geo origin file if the data is valid?

Reply:
Yes, it would be better to put these checks in `read_local_geo_cs_from_file`.  As a starter, we should initialize those local `lat, lon, alt` variables to `0.0`.  That is probably the cause of the non-repeatable behavior when an empty file was used.  We can also do something like `bool valid = ifs >> lat >> lon >> alt;` to check if the read was successful and return false or thrown an exception if not.  We should least have a failure/exception if the file is not opened or not read.  We could also consider checking if `lat` and `lon` are in valid ranges, but that's not as critical.  It would be good to add a unit test for this function that tries to load an empty file and test that it fails.

It might be best to make changes to `read_local_geo_cs_from_file` and related tests in a different PR.